src/webgpu/util/check_contents.ts

```diff
@@ -44,7 +44,28 @@ export function checkElementsEqual(
 ): ErrorWithExtra | undefined {
   assert(actual.constructor === expected.constructor, 'TypedArray type mismatch');
   assert(actual.length === expected.length, 'size mismatch');
-  return checkElementsEqualGenerated(actual, i => expected[i]);
```

Comment:
Hmmm... I would be interested to see if the change here where instead of creating an additional generator, we are now just comparing the arrays is the real culprit of the speed up? (Sort of the only thing that seems to really have changed functionally?)

Reply:
I measured these changes separately - see the commit messages. The largest speedup was from just splitting the check function in half.